### PR TITLE
[IMP] Added prometheus-exporter capabilities & nginx cache customization capabilities

### PR DESCRIPTION
**Pull request by wouitmil**

### CODE REVIEW

File: 9.0/Dockerfile [hunk 1]
```diff
@@ -9,6 +9,12 @@ COPY conf.d /etc/confd/conf.d
 COPY templates /etc/confd/templates
 COPY docker-entrypoint.sh /docker-entrypoint.sh
 
+WORKDIR /tmp
+RUN wget https://github.com/martin-helmich/prometheus-nginxlog-exporter/releases/download/v1.10.0/prometheus-nginxlog-exporter_1.10.0_linux_arm64.tar.gz
```
Comment (vrenaville) [inline 9.0/Dockerfile:13]: can we fork this repo into camptocamp org ?

File: 9.0/templates/nginx.conf.tmpl [hunk 2]
```diff
@@ -93,10 +102,12 @@ http {
 
     client_max_body_size 1G;
 
-    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+    add_header Strict-Transport-Security "max-age=600; includeSubDomains" always;
```
Comment (vrenaville) [inline 9.0/templates/nginx.conf.tmpl:105]: 5TxEffER5vjrPe4This part was for test, I think it should be revert to original when we will merge to master

File: 9.0/templates/nginx.conf.tmpl [hunk 3]
```diff
@@ -125,7 +136,7 @@ http {
       auth_basic_user_file /etc/nginx/htpasswd;
       {{ end }}
       add_header X-Static no;
-      add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+      add_header Strict-Transport-Security "max-age=600; includeSubDomains" always;
```
Comment (vrenaville) [inline 9.0/templates/nginx.conf.tmpl:139]: same here

File: 9.0/templates/nginx.conf.tmpl [hunk 4]
```diff
@@ -173,7 +192,7 @@ http {
 
       proxy_cache_bypass $http_cache_control;
       add_header X-Cache-Status $upstream_cache_status;
-      add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+      add_header Strict-Transport-Security "max-age=600; includeSubDomains" always;
       # there is no inheritance of proxy_set_header, as soon as we define one at a level,
```
Comment (vrenaville) [inline 9.0/templates/nginx.conf.tmpl:196]: same here

**Comment by vrenaville** [comment]
@wouitmil did you have test this image on our test instance, do we know if it has latency due to log writing
 ?

**Review by vrenaville — CHANGES_REQUESTED** [review]
I would like to have this branch devel tested first :) on a customer instance.

we will need to adapt the helm package to add servicemonitore entry point

File: 9.0/templates/nginx.conf.tmpl [hunk 5]
```diff
@@ -77,7 +86,7 @@ http {
   proxy_max_temp_file_size  2048m;
   proxy_temp_file_write_size  64k;
 
-  proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=one:10m inactive=60m;
+  proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=one:100m inactive=60m;
```
Comment (gurneyalex) [inline 9.0/templates/nginx.conf.tmpl:89]:
```suggestion
  # 20MB of shared memory for cache keys, inactivity delay 1h before forced purge (do we need this?) 
  proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=one:100m inactive=1h;
```

Documentation for the directive is on: https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path

Now, reading the doc: keys_zone at 10MB is probably enough, 100MB is very likely way too much

> all active keys and information about data are stored in a shared memory zone, whose name and size are configured by the keys_zone parameter. One megabyte zone can store about 8 thousand keys. 

so 10M gives 80k cached entries, maybe we want to raise this to 20M but 100M  sounds really high. 

`max_size` is a better parameter if we want to control the on disk space (or `min_free` if we only want to control the free amount of disk we want to keep -> this is likely tied to the use of a docker volume -> to check what space we have (and ensure /var/cache/nginx is on a volume, and not on the overlayfs).

File: 9.0/templates/config.hcl.tmpl [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    files = [
+      "/var/log/nginx/access_prometheus.log",
+    ]
+  }
+
+  format = "$remote_addr - $remote_user [$time_local] \"$request\" \"$request_id\" $status \"$upstream_cache_status\" $body_bytes_sent \"$http_referer\" \"$http_user_agent\" \"$http_x_forwarded_for\" rt=$request_time uct=\"$upstream_connect_time\" uht=\"$upstream_header_time\" urt=\"$upstream_response_time\""
+  
+  labels {
+    app = "odoo"
+    website = "ocms"
```
Comment (simahawk) [inline 9.0/templates/config.hcl.tmpl:21]: ocms?

**Pull request closed without merging**